### PR TITLE
Fix a few bugs and warnings

### PR DESCRIPTION
**Pull request by shika-blyat**

### CODE REVIEW

**Comment by Astremy** [comment]
Hello,

There are several wrong things.

First of all, your commit rewrites all the files you change, making it impossible to see what has actually changed.
Second, IntEnum is not useful, so I think you might as well take it out.

Finally, the Perm.permission == ?? to test if it includes the permission doesn't work anymore (indeed, for example 255 == Perm.ADMINISTRATOR should work (since the permission is included in it).

Please fix this and the conflict issues.

**Comment by shika-blyat** [comment]
Paz

File: piscord/Gateway.py [hunk 1]
```diff
@@ -2,110 +2,118 @@
...
+        self.url = url
+        self.token = token
+        self.auth_op = auth_op
+        self.last_sequence = 0
+        self.events_code = events_code
+        self.heartbeat_code = heartbeat_code
+        self.presence = presence
+        self.session_id = None
+        self.error = None
+        self.heartbeat = None
```
Comment (shika-blyat) [inline piscord/Gateway.py:20]: Add a few None initialization and move all the None initialization at the bottom of the __init__ method

File: piscord/Gateway.py [hunk 2]
```diff
@@ -2,110 +2,118 @@
...
+    async def connect(self, action=None):
+        self.loop = asyncio.get_event_loop()
+        ws = await websockets.connect(self.url, ping_interval=None)
+        self.ws = ws
+        while True:
+            if self.error:
+                raise self.error
+            try:
+                msg = await self.ws.recv()
+            except websockets.exceptions.ConnectionClosed as e:
```
Comment (shika-blyat) [inline piscord/Gateway.py:31]: Name the Exception that the except block is supposed to catch

File: piscord/Gateway.py [hunk 3]
```diff
@@ -2,110 +2,118 @@
...
+                        "presence": self.presence
+                    }}
+                if action:
+                    await self.send(action)
+                else:
+                    await self.send(payload)
+            if data["op"] == self.events_code:
+                if data["t"] == "READY":
+                    self.session_id = data["d"]["session_id"]
+            yield data
```
Comment (shika-blyat) [inline piscord/Gateway.py:84]: Refactor everything according to the new file you gave me

File: piscord/Gateway.py [hunk 4]
```diff
@@ -2,110 +2,118 @@
...
+            }}
+        ws = await websockets.connect(self.url, ping_interval=None)
+        msg = None
+        try:
+            await ws.recv()
+            await self.send(payload)
+            msg = await ws.recv()
+        except:
+            await ws.close()
+        return ws, msg
```
Comment (shika-blyat) [inline piscord/Gateway.py:102]: Same thing, that was your changes

File: piscord/Gateway.py [hunk 5]
```diff
@@ -2,110 +2,118 @@
...
+    async def _stop(self):
+        self.session_id = None
+        await self.ws.close()
+        self.heartbeat.cancel()
 
-	def stop(self):
-		asyncio.run_coroutine_threadsafe(self._stop(), self.loop)
\ No newline at end of file
+    def stop(self):
+        asyncio.run_coroutine_threadsafe(self._stop(), self.loop)
```
Comment (shika-blyat) [inline piscord/Gateway.py]: just indentation changes

File: piscord/OAuth.py [hunk 6]
```diff
@@ -1,48 +1,49 @@
-from .API_Elements import *
+from .API_Elements import Member, Guild, User
```
Comment (shika-blyat) [inline piscord/OAuth.py:1]: Make imports explicit

File: piscord/OAuth.py [hunk 7]
```diff
@@ -1,48 +1,49 @@
...
+
+    def get_guilds(self, token):
+        if "guilds" in self.scope:
+            return [Guild(guild, self.bot) for guild in self.__request_token(token, "/users/@me/guilds")]
+        return "Invalid Scope"
+
+    def add_guild_member(self, token, guild_id, user_id):
+        if "guilds.join" in self.scope:
+            return Member({**self.bot.api_call(f"/guilds/{guild_id}/members/{user_id}", "PUT", json=json.loads(token)["access_token"]), "guild_id": guild_id}, self.bot)
+        return "Invalid Scope"
```
Comment (shika-blyat) [inline piscord/OAuth.py:49]: Same again

File: piscord/Voice.py [hunk 8]
```diff
@@ -1,95 +1,92 @@
...
+            if data["op"] == 0:
+                if data["t"] == "READY":
+                    await gateway.send({
+                        "op": 4,
+                        "d": {
+                            "guild_id": self.guild_id,
+                            "channel_id": self.channel_id,
+                            "mute": self.mute,
+                            "deaf": self.deaf
+                        }})
```
Comment (shika-blyat) [inline piscord/Voice.py:34]: Also your changes

File: piscord/Voice.py [hunk 9]
```diff
@@ -1,95 +1,92 @@
...
+                payload = {
+                    "op": 1,
+                    "d": {
+                        "protocol": "udp",
+                        "data": {
+                            "address": self.ip,
+                            "port": self.port,
+                            "mode": "xsalsa20_poly1305_lite"
+                        }}}
+                await gateway.send(payload)
```
Comment (shika-blyat) [inline piscord/Voice.py:89]: Your changes again

**Pull request closed without merging**